Client/Impl/Worker/JobWorker.cs

```diff
@@ -119,6 +133,62 @@ internal void Open()
             activateJobsRequest.Type);
     }
 
+    private async Task StreamJobs(ITargetBlock<IJob> input, CancellationToken cancellationToken)
+    {
+        while (!source.IsCancellationRequested)
+        {
+            try
+            {
+                logger?.LogDebug(
+                    "Job worker stream ({worker}) for job type {type} is starting.",
+                    activateJobsRequest.Worker,
+                    activateJobsRequest.Type);
+
+                var stream = jobActivator.SendStreamActivateRequest(streamActivateJobsRequest);
+
+                logger?.LogDebug(
+                    "Job worker stream ({worker}) for job type {type} has been opened.",
+                    activateJobsRequest.Worker,
+                    activateJobsRequest.Type);
+
+                while (!source.IsCancellationRequested)
+                {
+                    var currentJobs = Thread.VolatileRead(ref currentJobsActive);
+
+                    if (currentJobs >= thresholdJobsActivation)
+                    {
+                        await Task.Delay(StreamBackpressureDelayMs, cancellationToken);
+                        continue;
+                    }
+
+                    if (!await stream.ResponseStream.MoveNext(cancellationToken))
+                    {
+                        logger?.LogDebug("Job stream MoveNext returned false; retrying shortly");
+                        await Task.Delay(StreamNoDataDelayMs, cancellationToken);
+                        continue;
+                    }
+
+                    var grpcActivatedJob = stream.ResponseStream.Current;
+                    var activatedJob = new Zeebe.Client.Impl.Responses.ActivatedJob(grpcActivatedJob);
+                    var response = new Zeebe.Client.Impl.Responses.ActivateJobsResponses();
+                    response.Jobs.Add(activatedJob);
+
+                    logger?.LogDebug(
+                        "Job stream: new task received for worker {worker} with key {key}",
+                        grpcActivatedJob.Worker,
+                        grpcActivatedJob.Key);
+
+                    await HandleActivationResponse(input, response, null);
```

Comment:
🔧 I think we could change the signature of the method to accept a list of jobs, then we wouldn't need to unnecessarily create a response object.